v11y/api/src/db/mod.rs

```diff
@@ -382,13 +407,92 @@ impl Db {
                 .map(|row| (row.get::<String, _>("origin"), row.get::<String, _>("related")))
         })
     }
+
+    pub async fn get_references<'s>(
+        &'s self,
+        id: &'s str,
+        r#type: Option<String>,
+        origin: Option<String>,
+    ) -> impl Stream<Item = (String, Reference)> + 's {
+        let query = if let Some(origin) = origin {
+            if let Some(ty) = r#type {
+                sqlx::query(
+                    r#"
+                select
+                    origin, type, url
+                from
+                    refs
+                where
+                    vulnerability_id = $1 and origin = $2 and type = $3
+                "#,
+                )
+                .bind(id)
+                .bind(origin)
+                .bind(ty)
+            } else {
+                sqlx::query(
+                    r#"
+                select
+                    origin, type, url
+                from
+                    refs
+                where
+                    vulnerability_id = $1 and origin = $2
+                "#,
+                )
+                .bind(id)
+                .bind(origin)
+            }
+        } else if let Some(ty) = r#type {
+            sqlx::query(
+                r#"
+                select
+                    origin, type, url
+                from
+                    refs
+                where
+                    vulnerability_id = $1 and type = $2
+                order by
+                    origin
+                "#,
+            )
+            .bind(id)
+            .bind(ty)
+        } else {
+            sqlx::query(
+                r#"
+                select
+                    origin, type, url
+                from
+                    refs
+                where
+                    vulnerability_id = $1
+                order by
+                    origin
+                "#,
+            )
+            .bind(id)
+        };
+
+        query.fetch(&self.pool).filter_map(|row| async move {
+            row.ok().map(|row| {
+                (
+                    row.get::<String, _>("origin"),
```

Comment:
Why is `row.get::<String, _>("origin")` better than `row.get("origin")`? The compiler doesn't seem to be requiring that?

Reply:
Ah, I think a vestige of my previous testing where I wasn't giving it enough information to infer the type binding.

Lemme revisit.